src/taipy/gui/renderers/builder.py

```diff
@@ -464,44 +464,40 @@ def __update_col_desc_from_indexed(self, columns: t.Dict[str, t.Any], name: str)
             else:
                 warnings.warn(f"{self.__element_name} {name}[{k}] is not in the list of displayed columns")
 
-    def _get_dataframe_attributes(self, date_format="MM/dd/yyyy", number_format=None):  # noqa: C901
-        data = self.__attributes.get("data")
-        data_hash = self.__hashes.get("data", "")
-        col_types = self.__gui._accessors._get_col_types(data_hash, _TaipyData(data, data_hash))
-        columns = _get_columns_dict(
-            data,
-            _add_to_dict_and_get(self.__attributes, "columns", {}),
-            col_types,
-            _add_to_dict_and_get(self.__attributes, "date_format", date_format),
-            _add_to_dict_and_get(self.__attributes, "number_format", number_format),
-        )
+    def _get_dataframe_attributes(self, date_format="MM/dd/yyyy", number_format=None) -> '_Builder': # noqa: C901
+        columns, col_types, reset_fn_hash = self.__gui._calculate_table_columns(_is_boolean_true(t.cast(bool, self.__attributes.get("reset", False))), self.__hashes.get("reset"),
```

Comment:
Couldn't spot where _calculate_table_columns() is defined... strange

Reply:
in Gui